convert/convert.py

```diff
@@ -1,35 +1,66 @@
+import re
+import subprocess
+
 import discord
-from pint import UnitRegistry
 from redbot.core import commands
 
 
 class Convert(commands.Cog):
     """Convert related commands."""
 
     def __init__(self):
-        self.__ureg = UnitRegistry()
+        self.decimal = re.compile(r"\d+")
 
     @commands.command()
-    async def convert(self, ctx, *unit):
+    async def convert(self, ctx, *, conversion: str):
         """Convert from different kinds of units to other units using pint
-
         Example:
         - `[p]convert <from> to <to>`
         - `[p]convert 23cm to in`
         - `[p]convert 5in + 5ft to cm`
         """
 
+        if " to " not in conversion:
+            await ctx.send(
+                f"`{conversion}` is not a valid conversion. Please make sure it is in the format `[p]convert <from> to <to>`"
+            )
+            await ctx.send_help()
+            return
+
+        arg1, end_unit = conversion.split(" to ")
+
+        amount = re.search(self.decimal, arg1).group()
```

Comment:
```suggestion
        amount = self.decimal.search(arg1).group()
```
We can use search directly on the pattern